encode.go

```diff
@@ -303,12 +303,18 @@ func (p *Part) encodeContentFromReader(b *bufio.Writer) error {
 }
 
 // selectTransferEncoding scans content for non-ASCII characters and selects 'b' or 'q' encoding.
-func selectTransferEncoding(content []byte, quoteLineBreaks bool) transferEncoding {
+func (p *Part) selectTransferEncoding(content []byte, quoteLineBreaks bool) transferEncoding {
 	if len(content) == 0 {
```

Comment:
```suggestion
	if len(content) == 0 || p.encoder != nil && p.encoder.enforceQuotedCte {
```
I think most of the changes you made to this func are not needed, we don't care about a threshold if we are always selecting te7Bit?